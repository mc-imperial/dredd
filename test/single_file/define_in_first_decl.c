#define TYPE int

TYPE foo() {
  return 1 + 2;
}
