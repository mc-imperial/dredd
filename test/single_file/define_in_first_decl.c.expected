#define TYPE int

#include <inttypes.h>
#include <stdlib.h>
#include <string.h>
static int __dredd_enabled_mutation(int local_mutation_id) {
  static int initialized = 0;
  static uint64_t enabled_bitset[1];
  if (!initialized) {
    const char* dredd_environment_variable = getenv("DREDD_ENABLED_MUTATION");
    if (dredd_environment_variable) {
      char* temp = malloc(strlen(dredd_environment_variable) + 1);
      strcpy(temp, dredd_environment_variable);
      char* token;
      token = strtok(temp, ",");
      while(token) {
        int value = atoi(token);
        int local_value = value - 0;
        if (local_value >= 0 && local_value < 7) {
          enabled_bitset[local_value / 64] |= (1 << (local_value % 64));
        }
        token = strtok(NULL, ",");
      }
      free(temp);
    }
    initialized = 1;
  }
  return enabled_bitset[local_mutation_id / 64] & (1 << (local_mutation_id % 64));
}

static int __dredd_replace_binary_operator_Add_int_int(int arg1, int arg2, int local_mutation_id) {
  if (__dredd_enabled_mutation(local_mutation_id + 0)) return arg1 / arg2;
  if (__dredd_enabled_mutation(local_mutation_id + 1)) return arg1 * arg2;
  if (__dredd_enabled_mutation(local_mutation_id + 2)) return arg1 % arg2;
  if (__dredd_enabled_mutation(local_mutation_id + 3)) return arg1 - arg2;
  if (__dredd_enabled_mutation(local_mutation_id + 4)) return arg1;
  if (__dredd_enabled_mutation(local_mutation_id + 5)) return arg2;
  return arg1 + arg2;
}

TYPE foo() {
  if (!__dredd_enabled_mutation(6)) { return __dredd_replace_binary_operator_Add_int_int(1, 2, 0); }
}
